requirements.txt

```diff
@@ -29,7 +29,7 @@ PyYAML==5.1.2
 tenacity==6.0.0
 pyotp==2.3.0
 pexpect==4.7.0
-
+pytest-lazy-fixture==0.6.3
```

Comment:
What does it do, why do we need it and has this addition been discussed already?
I'm always a little cautious about adding new dependencies. We shouldn't add them lightly.

Also, looks like it's reason for git conflict.

Reply:
I posted my question in global Satellite Automation chat no one responded on that, @mshriver decide if he has better solution here. I gave it try and time and ended here https://github.com/pytest-dev/pytest/issues/349. Pytest currently does not support parameterize fixtures. Hence I ended using` pytest-lazy-fixture`. Please let if there is any better solution rather using this dependency I open to do that.

Reply:
Was this the question about indirect fixture parametrization? There should be other ways to handle this, without the additional dependency. Let's discuss further before proceeding with this.

Reply:
@mshriver Do you have any better solution other than I mentioned in the code?

Reply:
I don't think parametrizing the auth fixture function is necessary here.  You could parametrize the authentication method, indirectly, and combine the `auth_source` and `auth_source_ipa` fixtures into one method.  Based on the indirectly parametrized auth source type, the `auth_source` fixture would configure either AD/LDAP/IPA/etc.

Reply:
Can you please provide an example here with the current situation?

Reply:
@mshriver I'm not sure but are you talking about similar concepts mentioned here https://alysivji.github.io/pytest-fixures-with-function-arguments.html

Reply:
```
@pytest.fixture
def auth_source(request):
    if request.param.lower() == 'ldap':
        # entity create with ldap settings
        pass
    if request.param.lower() == 'ad':
        # entity create with AD settings
        pass
    else:
        # default auth server settings
        pass
    # read settings for the given auth, return that
    yield (auth_user, auth_password)
    # reset auth settings
    pass


@pytest.mark.parametrize("auth_source", ["LDAP", "AD"], indirect=True)
def test_login_failure(auth_source):
    user, password = auth_source
    # test content
```

Reply:
That's one approach, there are options with how username/password are parametrized as well. But the user/password that the test will use are directly related to which auth server was read from the settings and configured as auth provider, so I chose to yield them from the auth_source fixture instead of parametrizing them separately.

Reply:
The example that you linked (looks like it has some syntax errors, like _calling_ `pytest.fixture())` just returns a function from the fixture. This delays execution, and will actually run the auth server setup during test CALL phase, instead of test SETUP phase. Ideally fixtures execute their code during SETUP.

Reply:
I will try your solution and see if that works. Will let you soon.

Reply:
So... is this still required? I'm not against the dependency but I don't see any conclusion here.

Reply:
It is resolved and not required. I'm using @mshriver indirect parameterization

Reply:
Cool. Why is it still in the PR then?

Reply:
Yeah I cleaned it up